models/comment_model.js

```diff
@@ -11,4 +12,23 @@ function listComments(article_id) {
   })
 }
 
-module.exports = listComments
+function insertComment(article_id, commentData) {
+  const {author, body} =  commentData
+
+  return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+  .then((article) => {
+    if (article.rows.length === 0) {
+      return Promise.reject(
+        {status: 404, msg: 'Article not found'})
+    } else {
+      return db.query(`INSERT INTO comments (author, article_id, body) VALUES
+        ($1, $2, $3)
+        RETURNING *;`, [author, article_id, body]
+      ).then((result) => {
+        return result.rows[0];
+      })
```

Comment:
This is a good _INSERT_ query however it shouldn't be necessary to use a query to _SELECT_ from articles because it is a foreign key constraint PSQL will throw an error on the _INSERT_ query that you can handle 👍